src/main/java/mp4/Mp4Controller.java

```diff
@@ -0,0 +1,119 @@
+package mp4;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * The mp4 controller class for serving the reqeust.
+ * @author domi
+ *
+ */
+@RestController
+public class Mp4Controller {
+	private final static Logger log = LoggerFactory.getLogger(Mp4Controller.class);
+	private static Map<String, Object> cacheVMap = new HashMap<>();
+	private final static String cacheVMapKey_VMAP = "VMAP";
+	
+	
+	/**
+	 * Reload all videos from the mp4 media path.
+	 * @return
+	 */
+	@PostMapping("/medias/reload")
+	String reloadVideoList() {
+		cacheVMap.remove(cacheVMapKey_VMAP);
+		processVideoList();
+		log.info("reloaded medias");
+		return "reloaded";
+	}
+
+	/**
+	 * Get the file list json string for all videos.
+	 * @return
+	 */
+	@GetMapping("/medias")
+	String getPlayListJson() {
+		return processVideoList().get("playListJson");
+	}
+
+	/**
+	 * Get the video link menu html content.
+	 * @return
+	 */
+	@GetMapping("/medias/links")
+	String getListOfFilesLink() {
+		return processVideoList().get("videoLinkMenu");
```

Comment:
the key videoLinkMenu can be extract as constant.

Reply:
done with 022c320d16534f1c78695919d46b4180283a55d4